src/main/java/org/rumbledb/context/Name.java

```diff
@@ -0,0 +1,214 @@
+ * (e.g., name of the function or variable in JSONiq code).
+ * 
+ * Expanded names serve as (global or local) variable names, function names, and error codes.
+ * 
+ * @author Ghislain Fourny
+ *
+ */
+public class Name implements Comparable<Name>, Serializable, KryoSerializable {
+    private static final long serialVersionUID = 1L;
+    private String namespace;
+    private String prefix;
+    private String localName;
+    private static final String RUMBLE_NS = "http://rumbledb.org/main-namespace";
+    public static final Name CONTEXT_ITEM = createVariableInNoNamespace("$");
+    public static final Name CONTEXT_POSITION = createVariableInNoNamespace("$position");
+    public static final Name CONTEXT_COUNT = createVariableInNoNamespace("$count");
+
+    private Name() {
+        this.namespace = null;
+        this.prefix = null;
+        this.localName = null;
+    }
+
+    private Name(String namespace, String prefix, String localName) {
+        this.namespace = namespace;
+        this.prefix = prefix;
+        this.localName = localName;
+        if (this.prefix != null && this.namespace == null) {
+            throw new OurBadException("Namespace is null, but prefix is present");
+        }
+    }
+
+    /**
+     * Creates an expanded name that has no namespace. In JSONiq, unprefixed variables are never in
+     * a namespace.
+     * 
+     * @param localName the name of the variable
+     * @return the expanded name
+     */
+    public static Name createVariableInNoNamespace(String localName) {
+        return new Name(null, null, localName);
+    }
+
+    /**
+     * Creates an expanded name that has the Rumble namespace. By default, in Rumble, unprefixed
+     * function names live in the Rumble namespace. This namespace is for convenience and includes
+     * all functions in XQuery's fn namespace, JSONiq's jn and jnlib namespaces, as well as any
+     * user-defined functions with unprefixed names.
+     * 
+     * @param localName the name of the variable
+     * @return the expanded name
+     */
+    public static Name createVariableInRumbleNamespace(String localName) {
+        return new Name(RUMBLE_NS, "", localName);
+    }
+
+    /**
+     * Converts the expanded name to an expanded name whose local name has a #n suffix, where n is the arity of the
+     * function,
+     * in order to encode the whole function identifier as an expanded name.
+     * This is only used to keep track of dependencies between functions and variables in the dependency
+     * resolution mechanism, and it should not be used for other purposes.
+     * 
+     * @param arity the arity of the function.
+     * @return the expanded name
+     */
+    public Name addArityToFunctionName(int arity) {
+        return new Name(this.namespace, this.prefix, this.localName + "#" + arity);
+    }
+
+    /**
+     * Returns the namespace.
+     * 
+     * @return the namespace, or null.
+     */
+    public String getNamespace() {
```

Comment:
I think these getter comments can be left out.

Reply:
Done